src/app/contexts/RequestContext/getMetaUrls/index.js

```diff
@@ -0,0 +1,35 @@
+const getAmpUrl = url => `${url}.amp`;
+
+const getCanonicalUrl = (origin, pathname) => {
+  const ampRegex = /(.amp)/g;
+  const canonicalUrl = `${origin}${pathname}`;
+
+  return canonicalUrl.replace(ampRegex, '');
+};
+
+const getUkCanonicalUrl = url => {
+  const tldRegex = /(.com|.co.uk)/g;
+  return url.replace(tldRegex, '.co.uk');
+};
+
+const getNonUkCanonicalUrl = url => {
```

Comment:
This doesn't necessarily get the canonical URL, you could equally pass the amp url and get the non-UK version of that (similarly for the UK function above).

```suggestion
const getNonUkUrl = url => {
```

Reply:
> you could equally pass the amp url

The way this file takes `origin, pathname`, and that we have `getAmpUrl` above, this is more accurate as 'Canonical' IMO, thoughts?

Reply:
To me the function name `getNonUkCanonicalUrl(url)` suggests it'll convert any URL into the non-UK canonical version, but as you say given the way this file is laid out it's fairly explanatory that it won't